service/worker/scheduler/workflow.go

```diff
@@ -817,7 +819,10 @@ func (s *scheduler) processWatcherResult(id string, f workflow.Future) {
 	if idx := slices.IndexFunc(s.Info.RunningWorkflows, match); idx >= 0 {
 		s.Info.RunningWorkflows = slices.Delete(s.Info.RunningWorkflows, idx, idx+1)
 	} else {
-		s.logger.Error("closed workflow not found in running list", "workflow", id)
+		// This could happen if the watcher activity gets interrupted and is retried after the
+		// heartbeat timeout, but in the meantime we have done a refresh through a local activity.
+		// This often happens when the worker is restarted.
+		s.logger.Warn("closed workflow not found in running list", "workflow", id)
```

Comment:
Seeing this message in the logs is very confusing.

> Of course, "closed workflow" is not the "running list". Why it is a warning?

Reply:
I can reword (maybe in another PR):

This is the point at which workflows that have just closed are removed from the running list. So for one that just closed, we expect to find it here. But as the new comment suggests, it's possible that another part of the workflow already removed it, so we wouldn't find it a second time.